### DISCUSSION TITLE
samples/mqtt_azure: descriptor not set on non secure transport

### DISCUSSION THREAD

**Discussion by fgervais** [zephyrproject-rtos/zephyr#63455] [Q&A] [answered]
In the `mqtt_azure` sample, the socket descriptor is only set when using secure mqtt transport:

https://github.com/zephyrproject-rtos/zephyr/blob/026dbdf5392ea0cb96ff8a9a204a2d239de096a5/samples/net/cloud/mqtt_azure/src/main.c#L91

What happens then when not using the secure transport?
On what are we waiting for in the `wait()` function?

**Comment by kartben** [comment, marked as answer]
mmmm this doesn't make much sense to have an if there since it's always going to be a secure connection (https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/cloud/mqtt_azure/src/main.c#L179, and IoT Hub doesn't support non-TLS anyway)
Did you want to maybe submit a patch?

**Reply by fgervais** [reply]
Thank you for your input.

I took a note, if I have an opening I send a PR for this.
